### PR TITLE
Add support for subclassing gio::Application

### PR DESCRIPTION
**Pull request by sdroege**
Based on code by Dirk Van Haerenborgh @vhdirk from https://github.com/gtk-rs/gobject-subclass

### CODE REVIEW

File: src/subclass/application.rs [hunk 1]
```diff
@@ -0,0 +1,529 @@
+// Copyright 2019, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+use gio_sys;
```
Comment (GuillaumeGomez) [inline src/subclass/application.rs]: You have the right to add an empty line before this one. :)

Comment (sdroege) [reply src/subclass/application.rs]: done

File: src/subclass/application.rs [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+use ApplicationClass;
+
+use libc::{c_char, c_void};
+use std::convert;
+use std::ffi::OsString;
+use std::fmt;
+use std::ops::Deref;
+use std::ptr;
+
+pub struct ArgumentList {
```
Comment (EPashkin) [inline src/subclass/application.rs]: Why not move `ArgumentList` to separate file?

File: src/subclass/application.rs [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+
+impl fmt::Debug for ArgumentList {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        self.items.fmt(formatter)
+    }
+}
+
+impl convert::Into<Vec<OsString>> for ArgumentList {
+    fn into(self) -> Vec<OsString> {
+        self.items.clone()
```
Comment (EPashkin) [inline src/subclass/application.rs]: Clone really needed? Why not replace it with empty vector?

**Comment by EPashkin** [comment]
@sdroege Thanks

File: src/subclass/application.rs (hunk 2 above)
Comment (sdroege) [reply src/subclass/application.rs]: Because it will only ever be used inside this file

File: src/subclass/application.rs (hunk 3 above)
Comment (sdroege) [reply src/subclass/application.rs]: Indeed it should be possible to just move it out

Comment (sdroege) [reply src/subclass/application.rs]: Fixed. Thanks :)

**Comment by EPashkin** [comment]
@sdroege 👍

**Pull request merged**